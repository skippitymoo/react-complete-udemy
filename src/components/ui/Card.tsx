import React, { JSX } from 'react';

export const Card = ({
  className,
  children,
}: React.ComponentPropsWithoutRef<'div'>): JSX.Element => {
  return <div className={`card ${className ?? ''}`.trim()}>{children}</div>;
};
