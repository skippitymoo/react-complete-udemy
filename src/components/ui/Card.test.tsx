import { render } from '@testing-library/react';
import { Card } from './Card';

describe('Card', () => {
  it('should render the class(es) and children provided', () => {
    const { container } = render(
      <Card className='test-class test-class__yeo'>
        <div>Yeooo</div>
        <div>Yeooo 2</div>
      </Card>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
