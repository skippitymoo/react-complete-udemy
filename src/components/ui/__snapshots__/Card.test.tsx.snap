// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render the class(es) and children provided 1`] = `
<div
  class="card test-class test-class__yeo"
>
  <div>
    Yeooo
  </div>
  <div>
    Yeooo 2
  </div>
</div>
`;
